be/src/exprs/vectorized/array_functions.cpp

```diff
@@ -107,6 +107,222 @@ ColumnPtr ArrayFunctions::array_append([[maybe_unused]] FunctionContext* context
+        }
+
+#define HANDLE_ELEMENT_TYPE(ElementType)                                                                          \
+    do {                                                                                                          \
+        if (typeid(*elements_ptr) == typeid(ElementType)) {                                                       \
+            return _process<NullableElement, NullableTarget, ConstTarget>(                                        \
+                    *down_cast<const ElementType*>(elements_ptr), array_offsets, *targets_ptr, null_map_elements, \
+                    null_map_targets);                                                                            \
+        }                                                                                                         \
+    } while (0)
+
+        HANDLE_ELEMENT_TYPE(BooleanColumn);
+        HANDLE_ELEMENT_TYPE(Int8Column);
+        HANDLE_ELEMENT_TYPE(Int16Column);
+        HANDLE_ELEMENT_TYPE(Int32Column);
+        HANDLE_ELEMENT_TYPE(Int64Column);
+        HANDLE_ELEMENT_TYPE(Int128Column);
+        HANDLE_ELEMENT_TYPE(FloatColumn);
+        HANDLE_ELEMENT_TYPE(DoubleColumn);
+        HANDLE_ELEMENT_TYPE(DecimalColumn);
+        HANDLE_ELEMENT_TYPE(Decimal32Column);
+        HANDLE_ELEMENT_TYPE(Decimal64Column);
+        HANDLE_ELEMENT_TYPE(Decimal128Column);
+        HANDLE_ELEMENT_TYPE(BinaryColumn);
+        HANDLE_ELEMENT_TYPE(DateColumn);
+        HANDLE_ELEMENT_TYPE(TimestampColumn);
+        HANDLE_ELEMENT_TYPE(ArrayColumn);
+
+        // TODO(zhuming): demangle class name
+        LOG(ERROR) << "unhandled column type: " << typeid(array_elements).name();
+        DCHECK(false) << "unhandled column type: " << typeid(array_elements).name();
+        return ColumnHelper::create_const_null_column(array_elements.size());
+    }
+
+    // array is non-nullable.
+    static ColumnPtr _array_remove_non_nullable(const ArrayColumn& array, const Column& arg) {
+        bool nullable_element = false;
+        bool nullable_target = false;
+        bool const_target = false;
+        ColumnPtr targets_holder;
+
+        const UInt32Column& offsets = array.offsets();
+        const Column* elements = &array.elements();
+        const Column* targets = &arg;
+        if (auto nullable = dynamic_cast<const NullableColumn*>(elements); nullable != nullptr) {
+            // If this nullable column does NOT contains any NULL, process it as non-nullable column.
+            nullable_element = nullable->has_null();
+            elements = nullable->has_null() ? elements : nullable->data_column().get();
+        }
+        if (auto nullable = dynamic_cast<const NullableColumn*>(targets); nullable != nullptr) {
+            nullable_target = nullable->has_null();
+            targets = nullable->has_null() ? targets : nullable->data_column().get();
+        }
+        if (targets->only_null() && !nullable_element) {
```

Comment:
What's this condition for?
When targets is only null and there is not a nullable element.
It will copy the input array and return other than a unint8column.